src/util/entropy_utils.js

```diff
@@ -0,0 +1,133 @@
+/* Copyright (C) 2025 NooBaa */
+'use strict';
+
+const fs = require('fs');
+const util = require('util');
+const chance = require('chance')();
+const child_process = require('child_process');
+const config = require('../../config');
+const blockutils = require('linux-blockutils');
+
+const DEVICE_TYPE_DISK = 'disk';
+
+const async_delay = util.promisify(setTimeout);
+const async_exec = util.promisify(child_process.exec);
+const get_block_info_async = util.promisify(blockutils.getBlockInfo);
+
+/**
+ * generate_entropy will create randomness by changing the MD5
+ * using information from the device (disk)
+ * it will run as long as the callback it true
+ * @param {function} loop_cond
+ */
+async function generate_entropy(loop_cond) {
+    if (process.platform !== 'linux' || process.env.container === 'docker') return;
+    while (loop_cond()) {
+        try {
+            await async_delay(1000);
+            const ENTROPY_AVAIL_PATH = '/proc/sys/kernel/random/entropy_avail';
+            const entropy_avail = parseInt(await fs.promises.readFile(ENTROPY_AVAIL_PATH, 'utf8'), 10);
+            console.log(`generate_entropy: entropy_avail ${entropy_avail}`);
+            if (entropy_avail >= config.ENTROPY_MIN_THRESHOLD) return;
+            const available_disks = await get_block_device_disk_info();
+            const disk_details = pick_a_disk(available_disks);
+            if (disk_details) {
+                await add_entropy(disk_details.name, disk_details.size);
+            } else {
+                throw new Error('No disk candidates found');
+            }
+        } catch (err) {
+            // we intentionally don't print the error
+            // as console.err/warm and the error with stack trace
+            // as the generate_entropy is our best effort to generate entropy
+            // until Linux will finally do it
+            console.log('generate_entropy: retrying');
+        }
+    }
+}
+
+/**
+ * get_block_device_disk_info
+ * (on Linux) will return array of disk devices
+ *            that their names starts with /dev/
+ *            and their size it a number in bytes
+ * (else) will return an empty array
+ * Note: under the hood it uses blockutils.getBlockInfo which calls lsblk command
+ *       we used the option onlyStandard for parsing "disk" and "part" entries
+ *       reference: https://github.com/mw-white/node-linux-blockutils
+ * @returns {Promise<object[]>}
+ */
+async function get_block_device_disk_info() {
+    const is_linux = process.platform === 'linux';
+    if (!is_linux) return [];
+
+    const block_devices = await get_block_info_async({ "onlyStandard": true });
+    if (!block_devices) return [];
+
+    const available_disks = [];
+    for (const block_device of block_devices) {
+        if (block_device.TYPE === DEVICE_TYPE_DISK) {
+            available_disks.push({
+                    name: `/dev/${block_device.NAME}`,
+                    size: Number(block_device.SIZE),
+                });
```

Comment:
fix spacing to align with the code level:
```suggestion
            available_disks.push({
                name: `/dev/${block_device.NAME}`,
                size: Number(block_device.SIZE),
            });
```